### DISCUSSION TITLE
USB3320 + STM32F4 High Speed

### DISCUSSION THREAD

**Discussion by Xelus22** [hathach/tinyusb#1749] [Q&A]
Hello!

I've just built a USB3320 + STM32F446, and I'm trying to get the ULPI working. The onboard USB FS PHY works great and all with tinyUSB but I dont see an example or any code for STM32F4 for USB HS OTG. If anyone has any pointers that would be great!

Thanks,
Xelus

**Comment by hathach** [comment]
check out the H743 eval board, which also use ULPI, you will need to set CFG_TUD_MAX_SPEED to highspeed and change rhport to port 1 (or init with tud_init(1) )

**Reply by Xelus22** [reply]
Are the H7 peripherals close enough to the F4? Or will I need to port all the LD scripts etc over as well? 

Sorry I'm quite new to such low level embedded programming.

**Reply by hathach** [reply]
it is only the reference for USB-related code only namely: configuration, clock, pin setup. For other mcu-specific work, you will need to look at ST manual
